cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_single_col.hpp

```diff
@@ -0,0 +1,154 @@
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/backend/common.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/selection/kselect_by_rows_base.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+// Performs k-selection for k == 1
+template <typename Float>
+class kselect_by_rows_single_col : public kselect_by_rows_base<Float> {
+public:
+    kselect_by_rows_single_col() {}
+    sycl::event operator()(sycl::queue& queue,
+                           const ndview<Float, 2>& data,
+                           std::int64_t k,
+                           ndview<Float, 2>& selection,
+                           ndview<std::int32_t, 2>& indices,
+                           const event_vector& deps) override {
+        return select<true, true>(queue, data, selection, indices, deps);
+    }
+    sycl::event operator()(sycl::queue& queue,
+                           const ndview<Float, 2>& data,
+                           std::int64_t k,
+                           ndview<Float, 2>& selection,
+                           const event_vector& deps) override {
+        ndarray<std::int32_t, 2> dummy;
+        return select<true, false>(queue, data, selection, dummy, deps);
+    }
+    sycl::event operator()(sycl::queue& queue,
+                           const ndview<Float, 2>& data,
+                           std::int64_t k,
+                           ndview<std::int32_t, 2>& indices,
+                           const event_vector& deps) override {
+        ndarray<Float, 2> dummy;
+        return select<false, true>(queue, data, dummy, indices, deps);
+    }
+
+private:
+    template <bool selection_out, bool indices_out>
+    sycl::event select(sycl::queue& queue,
+                       const ndview<Float, 2>& data,
+                       ndview<Float, 2>& selection,
+                       ndview<std::int32_t, 2>& indices,
+                       const event_vector& deps = {}) {
+        if (indices_out) {
+            ONEDAL_ASSERT(indices.get_shape()[0] == data.get_shape()[0]);
+            ONEDAL_ASSERT(indices.get_shape()[1] == 1);
+        }
+        if (selection_out) {
+            ONEDAL_ASSERT(selection.get_shape()[0] == data.get_shape()[0]);
+            ONEDAL_ASSERT(selection.get_shape()[1] == 1);
+        }
+        const auto sg_sizes = queue.get_device().get_info<sycl::info::device::sub_group_sizes>();
```

Comment:
Suggest to extract method and place it here: https://github.com/oneapi-src/oneDAL/blob/2bc78f2d2502fdd6dc3a547ab91f36cb3bccbbc8/cpp/oneapi/dal/backend/common.hpp